pkg/azure/provider.go

```diff
@@ -553,8 +555,7 @@ func RedactClientID(sensitiveString string) string {
 	return r.ReplaceAllString(sensitiveString, "$1##### REDACTED #####$3")
 }
 
-// getCertAndPrivKeyInPEMFormat returns the certificate and private key to be
-// written to file
+// getCertAndPrivKeyInPEMFormat returns the certificate and private key to be  written to file
 // cert and private key are returned when object type is "secret"
 func getCertAndPrivKeyInPEMFormat(value string) (string, error) {
 	return decodePKCS12(value, true, true)
```

Comment:
This is the only place that calls `decodePKCS12`. Is there a reason we want to allow conditionally set `getCert` and `getKey`?

Reply:
Yeah, its no longer required. Updated it!